`cargo test --no-default-features` fails

I tried running `cargo test --no-default-features` on `adler`  v0.2.3.  I got this failure:

```
   Compiling adler v0.2.3 (/tmp/cdtemp.oLUdJk/adler)
error[E0432]: unresolved import `std::io`
   --> src/lib.rs:154:14
    |
154 |     use std::io::BufReader;
    |              ^^ could not find `io` in `std`

error[E0425]: cannot find function `adler32_reader` in this scope
   --> src/lib.rs:206:28
    |
206 |             let real_sum = adler32_reader(&mut buf).unwrap();
    |                            ^^^^^^^^^^^^^^ not found in this scope

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `adler`.
```

Is this supposed to succeed?  I'm running the test on debian with system-packaged rustc 1.46.0 and cargo 1.42.1.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -151,7 +151,6 @@ pub fn adler32_reader<R: BufRead>(reader: &mut R) -> io::Result<u32> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::io::BufReader;
 
     #[test]
     fn zeroes() {
@@ -198,8 +197,10 @@ mod tests {
         assert_eq!(adler.checksum(), 0x8e88ef11); // from above
     }
 
+    #[cfg(feature = "std")]
     #[test]
     fn bufread() {
+        use std::io::BufReader;
         fn test(data: &[u8], checksum: u32) {
             // `BufReader` uses an 8 KB buffer, so this will test buffer refilling.
             let mut buf = BufReader::new(data);

